aws-java-sdk-core/src/main/java/com/amazonaws/retry/PredefinedBackoffStrategies.java

```diff
@@ -43,6 +43,13 @@
     /** Maximum retry limit.  Avoids integer overflow issues. **/
     private static final int MAX_RETRIES = 30;
 
+    private static final long exponentialDelayHelper(int retriesAttempted, int baseDelay, int maxBackoffTime) {
+      long potentialWait = (1L << retriesAttempted) * baseDelay;
+      return (retriesAttempted > MAX_RETRIES) ? maxBackoffTime :
+              (potentialWait < 0  ? maxBackoffTime :
+              Math.min(potentialWait, maxBackoffTime));
+    }
+
```

Comment:
I would prefer to stick to long here,  if you take the multiplier and multiply it by a baseDelay you will still have an overflow.
I would prefer to handle the overflow inside the helper method.

What do you think about this?